componentified-todo-list/safari/src/component/TaskApp.tsx

```diff
@@ -2,64 +2,60 @@ import React, {useEffect, useState} from "react";
 import NewTask from "./NewTask";
 import ActiveTaskList from "./ActiveTaskList";
 import DoneTaskList from "./DoneTaskList";
+import {fetchAllTodos, postNewTodo, updateTodo, deleteTodo} from '../service/todoService'
 
 const TaskApp: React.FC<any> = () => {
-    const [tasks, setTasks] = useState<TaskStruct[]>([])
-
-    const addNewItemTop = (subject: string) => {
-        const task: TaskStruct = {
-            id: Math.floor(Math.random() * 101),
-            subject: subject,
-            isDone: false
+    const [tasks, setTasks] = useState<TodoModel[]>([])
+
+    const addNewItem = (subject: string) => {
+        const task: TodoModel = {
+            id: "",
+            ownerId: 6,
+            title: subject,
+            checked: false
         };
 
-        setTasks([task, ...tasks])
+        postNewTodo(task)
+            .then(_ => fetchAllTodos())
```

Comment:
__[LOW]__  
This extra `fetchAllTodos` is redundant and can be removed. You just need to add the newly created to-do item to your current to-do items. Same goes for other handlers.